UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java

```diff
@@ -42,27 +44,26 @@
     private static final String EXTERNAL_MQTT_BROKER = "test.mosquitto.org";
     private static final String LOCALHOST = "10.0.2.2";
     private static final String MQTT_SERVER = "tcp://" + LOCALHOST + ":1883";
+
     private static final String THROTTLE_CONTROL = "/smartcar/control/speed";
     private static final String STEERING_CONTROL = "/smartcar/control/angle";
-    private static final String PARK = "/smartcar/control/park";
-    private static final String ODOMETER_LOG = "/smartcar/odometer";
-    private static final String SPEEDOMETER_LOG = "/smartcar/speedometer";
+    private static final String ODOMETER = "/smartcar/car/distance";
```

Comment:
Odometer topic seems to be changed in line 50. I am quite sure that odometer topic was "/smartcar/odometer" in sketch "smartCar.ino". You might wanna double check this. It might create a conflict in soon future. ;)

Reply:
@edizgenc2021 I purposefully changed this to match the topic being published from the car side/ arduino file  :)
This solves the bug and now properly displays the odometer data on the screen. With the previous topic, the Odometer was set to "0 m" because it wasn't subscribing to the correct topic. Now, both the UI and Smartcar files match topics.

FYI, I changed SPEEDOMETER too. It is now "smartcar/speed"

Reply:
Nice :)